### PR TITLE
Multiple choice answering

### PR DESCRIPTION
**Pull request by amEhlee**
## Why was this PR opened?
Multiple Choice Answering Screen task completed

## Brief Description
Page that will render a list of questions pulled from backend into a quiz format. Each question will have 4(?) options with a specific correct and incorrect options. Functionality not implemented yet just rendering.

## Unrelated Changes
None 

## UI Screenshots
image

## Link to Referenced Trello Cards
https://trello.com/c/nUctbEik

### CODE REVIEW

**Comment by maggiechew** [comment]
Hey @amEhlee, just running your branch I'm running into an error; we updated the naming conventions in the app/learning-buddy .env, app.config.js and authprovider files with [my last PR](https://github.com/new-idea-machine/os-lesson-delivery-system/pull/18). Folks who are up-to-date with this will run into issues; you might want to pull from main to your branch and get that up to date, just a thought anyways.
Screenshot 2023-07-09 113743
Screenshot 2023-07-09 113700

File: apps/Learning-Buddy/data/SampleTestData.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{
+  "questions": [
+    {
+      "question1": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Duis molestie nunc dictum sapien auctor, et porttitor metus ornare. Nulla euismod, mauris interdum congue mattis, massa justo placerat metus, vel tincidunt.",
+      "options": {
+        "Correct": "A",
```
Comment (maggiechew) [inline apps/Learning-Buddy/data/SampleTestData.json:6]: Most of the time I'd expect something like "Correct": "The Pyramids", "Incorrect": ["The Eiffel Tour", "The Great Wall", "The Leaning Tower of Piza"]. I'm guessing what you're suggesting is analogous to "Correct Answer: Y" "Incorrect Answers: DQR" and when you say todo radio buttons DYRQ you're simply saying that's the order the options would be presented in, correct? Just wasn't super clear.

File: apps/Learning-Buddy/navigation/AppStack.jsx [hunk 2]
```diff
@@ -17,6 +18,7 @@ export const AppStack = () => {
     >
       <Stack.Screen name='Home Screen' component={HomeScreen} />
       <Stack.Screen name='Question Form' component={QuestionForm} />
+      <Stack.Screen name="Sample Test" component={SampleTestScreen} />
```
Comment (maggiechew) [inline apps/Learning-Buddy/navigation/AppStack.jsx:21]: You could rename this to "Answering Screen"; we're going to use this screen and repurpose it with the JSON provided from another screen that navigates here, so no need to call this "sample" or "test".

File: apps/Learning-Buddy/screens/SampleTestScreen.jsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        <Text style={styles.pageTitle}>SAMPLE TEST</Text>
+        {SampleTestData.questions.map((e, i) => {
+          return (
+            <View style={localStyles.card} key={i}>
+              <Text>Question {i + 1}</Text>
+              <Text>{e.question1}</Text>
+              <Text>Correct Answer: {e.options.Correct}</Text>
+              <Text>Incorrect Answers: {e.options.Incorrect}</Text>
+              <Text>
+                TODO: Radio Buttons:{' '}
```
Comment (maggiechew) [inline apps/Learning-Buddy/screens/SampleTestScreen.jsx:36]: I think this story is arguably not complete until there are radio buttons representative of their options that users can interact with. I don't know if Muke has any mockups that show this right now, but we need some way for a user to select an option.

File: apps/Learning-Buddy/screens/SampleTestScreen.jsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+                {ShuffleShuffle(e.options.Correct, e.options.Incorrect)}
+              </Text>
+            </View>
+          );
+        })}
+      </ScrollView>
+      <View style={localStyles.footer}>
+        <FunctionOnPressBigButton
+          content={'Submit'}
+          onPress={() => {
```
Comment (maggiechew) [inline apps/Learning-Buddy/screens/SampleTestScreen.jsx:46]: I'm a bit torn on this; ideally I'd like it if you could find a way to pass along an array with their choices as well as the data you were originally provided with, and then another function could perform the math of evaluating it and sending results to the next screen. The button would also then ideally be disabled if not all questions had been answered (something like, if you had an array with all answers and its length is not equal to the number of questions you're generating, the button is disabled, else it's not disabled... maybe using a useEffect watching the length of that array?)

If for now you could update this button to pass along an array of answers, as well as the original object, to another screen (e.g. "Home Screen" for now, perhaps) that's probably close enough.

**Review by maggiechew — CHANGES_REQUESTED** [review]
A good start but I think some core functionality is missing (radio buttons, some functionality added to the button in gathering user answers and passing along the original questions) required to complete the task fully. You also might want to pull from main so your .env stuff is up to date. Let me know if you have any questions :)

File: apps/Learning-Buddy/data/SampleTestData.json (hunk 1 above)
Comment (amEhlee) [reply apps/Learning-Buddy/data/SampleTestData.json:6]: Yep exactly that. When I added sample correct and incorrect options I just left it as A, B, C, D to save time thinking of any actual questions. Ill add in a couple sample questions to better showcase this but yeah its exactly what you said. "Todo Radios" just implies what order they'll be rendered in.

File: apps/Learning-Buddy/navigation/AppStack.jsx (hunk 2 above)
Comment (amEhlee) [reply apps/Learning-Buddy/navigation/AppStack.jsx]: Done 🫠

**Review by maggiechew — APPROVED** [review]

**Review by yanliu1111 — APPROVED** [review]

**Pull request merged**